### DISCUSSION TITLE
Beeper output stops working after GPS lock in Inav 7.1.0

### DISCUSSION THREAD

**Discussion by wchnflr** [iNavFlight/inav#10117] [General]
## Current Behavior
Reddit thread posted by user u/HANENG18 that discovered this bug: https://www.reddit.com/r/fpv/comments/1d6agbk/inav_beeper_strange_behaviour_wgps/
At least 3 Redditors have confirmed this bug.

## Steps to Reproduce
1. Assign a TX switch to the Beeper function in Modes tab.
2. TX switch turns on/off the external beeper at power up, but before GPS lock is satisfied.
3. Once GPS lock is satisfied, the external beeper no longer sounds. The ESC motor beepers still work.
4. Reset or power cycle the FC - the external beeper works again before GPS lock is satisfied. External beeper works again if the satellite number drops below gps_min_sats.

## Expected behavior
The Beeper function will work regardless of GPS lock status.

## Suggested solution(s)
Free the beeper function from GPS lock status.

## Additional context
- FC Board name and vendor: Various, including:
1. Flywoo Goku GN405 Nano: target FLYWOOF405PRO
2. Flywoo Goku GN405S AIO: unofficial target FLYWOOF405S_AIO
3. Speedybee F405 v4: target SPEEDYBEEF405V4
- INAV version string: 7.1.0

**Comment by breadoven** [comment]
I probably don't have this issue using the current master build. I say probably because I thought I randomly wasn't getting the GPS ready beeps sometimes. However, when I checked the other day the beeper worked as expected.

You need to provide a Diff(s) to see if there's any common setting that might be causing the issue in your case.

**Reply by wchnflr** [reply]
Thank you for your reply. Are you asking for a diff of my CLI or diff of source code?

**Reply by breadoven** [reply]
Just a CLI Diff of the settings.

**Reply by wchnflr** [reply]
[INAV_7.1.0_cli_diff_all_EXPLORERLR4_20240609_172809.txt](https://github.com/user-attachments/files/15754438/INAV_7.1.0_cli_diff_all_EXPLORERLR4_20240609_172809.txt)

**Reply by wchnflr** [reply]
I took another look at the beeper.c code and found this:
```
#ifdef USE_GPS
void beeperGpsStatus(void)
{
    // if GPS fix then beep out number of satellites
    if (STATE(GPS_FIX) && gpsSol.numSat >= 5) {
        uint8_t i = 0;
        do {
            beep_multiBeeps[i++] = 5;
            beep_multiBeeps[i++] = 10;
        } while (i < MAX_MULTI_BEEPS && gpsSol.numSat > i / 2);

        beep_multiBeeps[i-1] = 50; // extend last pause
        beep_multiBeeps[i] = BEEPER_COMMAND_STOP;

        beeper(BEEPER_MULTI_BEEPS);    //initiate sequence
    } else {
        beeper(BEEPER_RX_SET);
    }
}
#endif
```
So I enabled MULTI_BEEPS in CLI:
```
# beeper multi_beeps
Enabled MULTI_BEEPS
```
And sure enough I get beeped the number of satellites when there's at least gps_min_sats. If it's less than gps_min_sats, I only get single beeps. I suppose it's not exactly a bug, but more a lack of documentation or a misunderstanding of beeper settings.

**Reply by breadoven** [reply]
OK, I finally understand what you're doing. You didn't mention you've disabled pretty well all the beeper modes. That's why providing a Diff helps for troubleshooting. Seems you've worked out what the issue is ... `MULTI_BEEPS` was disabled.

Documentation probably does need updating to clarify what this mode affects.

**Reply by wchnflr** [reply]
> Documentation probably does need updating to clarify what this mode affects.

I'll fork the documentation and give it a shot.

**Comment by wchnflr** [comment]
I did another test today where I enabled RX_SET, and left my TX beeper switch on. The external beeper sounds whenever the GPS satellites is less than 6. When it hits 6 or more, the external beeper stays silent.
Then I covered the GPS module with my fingers. As soon as the GPS satellite count drops below 6, the external beeper sounds. When it goes up to 6 or more, it goes silent. This change in external beeper happens in real time. Meanwhile, the ESC motor buzzers are beeping without interruption.

My CLI setting:
`# beeper
Disabled:  RUNTIME_CALIBRATION  HW_FAILURE  RX_LOST  RX_LOST_LANDING  DISARMING  ARMING  ARMING_GPS_FIX  BAT_CRIT_LOW  BAT_LOW  GPS_STATUS  ACTION_SUCCESS  ACTION_FAIL  READY_BEEP  MULTI_BEEPS  DISARM_REPEAT  ARMED  SYSTEM_INIT  ON_USB  LAUNCH_MODE  LAUNCH_MODE_LOW_THROTTLE  LAUNCH_MODE_IDLE_START  CAM_CONNECTION_OPEN  CAM_CONNECTION_CLOSED`

I looked at the gps and beeper source code, but nothing jumped out at me since I'm just a dumb hardware engineer.
